### PR TITLE
[new release] jsonxt (1.0.0)

### PR DESCRIPTION
**Pull request by stevebleazard**
Jsonxt - JSON parsers for files, strings and more

- Project page: <a href="https://github.com/stevebleazard/ocaml-jsonxt">https://github.com/stevebleazard/ocaml-jsonxt</a>
- Documentation: <a href="https://stevebleazard.github.io/ocaml-jsonxt/">https://stevebleazard.github.io/ocaml-jsonxt/</a>

##### CHANGES:

- Initial release

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
The tests segfault on ppc64:
```
# Test Successful in 0.120s. 488 tests run.
#     jxtester alias tests/core/run_test_suite_all (got signal SEGV)
```

File: packages/jsonxt/jsonxt.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+* Standard interfaces including Yojson compatibility"""
+maintainer: "Steve Bleazard <stevebleazard@googlemail.com>"
+authors: "Steve Bleazard <stevebleazard@googlemail.com>"
+license: "MIT"
+homepage: "https://github.com/stevebleazard/ocaml-jsonxt"
+doc: "https://stevebleazard.github.io/ocaml-jsonxt/"
+bug-reports: "https://github.com/stevebleazard/ocaml-jsonxt/issues"
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "2.0" & build}
```
Comment (kit-ty-kate) [inline packages/jsonxt/jsonxt.1.0.0/opam:36]: Dune does not support the `{build}` tag. See https://github.com/ocaml/dune/issues/2147

File: packages/jsonxt/jsonxt.1.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  "alcotest" {with-test}
+  "cmdliner" {with-test}
+  "core_kernel" {with-test}
+  "fmt" {with-test}
+  "ppx_sexp_conv" {with-test}
+]
+build: [
+  ["dune" "subst"] {pinned & dev}
```
Comment (kit-ty-kate) [inline packages/jsonxt/jsonxt.1.0.0/opam:44]:
```suggestion
  ["dune" "subst"] {dev}
```
What's the reason for putting the two?

File: packages/jsonxt/jsonxt.1.0.0/opam (hunk 1 above)
Comment (stevebleazard) [reply packages/jsonxt/jsonxt.1.0.0/opam:36]: Thanks for the info, I have fixed this in the next release

File: packages/jsonxt/jsonxt.1.0.0/opam (hunk 2 above)
Comment (stevebleazard) [reply packages/jsonxt/jsonxt.1.0.0/opam:44]: Fixed this in the next release, I am still getting up to speed on the opam/dune interaction as this is my first opam+dune release

**Comment by avsm** [comment]
ppc64 has sometimes strange requirements with C bindings. Restricting this to `arch != ppc64` is fine to merge to opam-repo

**Comment by stevebleazard** [comment]
> 
> 
> ppc64 has sometimes strange requirements with C bindings. Restricting this to `arch != ppc64` is fine to merge to opam-repo

I added 'available: [arch != "ppc64"]' to jsonxt.opam and a bunch of opam 2.1 compiles failed with m4 missing. It's not clear why

**Comment by kit-ty-kate** [comment]
> I added 'available: [arch != "ppc64"]' to jsonxt.opam and a bunch of opam 2.1 compiles failed with m4 missing. It's not clear why

Sorry that's totally my fault. I've fixed the issue and the rest seems all good. Thanks!

**Comment by kit-ty-kate** [comment]
@stevebleazard  Actually it seems I've merged that too fast. I've just tested locally but it looks like your package infinit-loops when using OCaml 4.11 here:
```
Running[210]: (cd _build/default && /home/kit_ty_kate/.opam/4.11/bin/ocamlc.opt -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I lib/.jsonxt.objs/byte -intf-suffix .ml -no-alias-deps -opaque -open Jsonxt__ -o lib/.jsonxt.objs/byte/jsonxt__Utilities.cmo -c -impl lib/utilities.ml)
```
I'll fix your package constraint but you might want to look into that.

**Comment by stevebleazard** [comment]
@kit-ty-kate Thanks, I will take a look at 4.11, I suspect I've seen this type of issue before

**Pull request merged**